### PR TITLE
Use spf13/cobra for docker build

### PR DESCRIPTION
**Pull request by vdemeester**
Moves the build command to `api/client/image/build.go` and use cobra  🐍.

/cc @dnephin @thaJeztah @LK4D4 @cpuguy83 

🐸

Signed-off-by: Vincent Demeester vincent@sbr.pm

### CODE REVIEW

File: api/client/image/build.go [hunk 1]
```diff
@@ -30,58 +30,88 @@ import (
...
+		tags:      opts.NewListOpts(validateTag),
+		buildArgs: opts.NewListOpts(runconfigopts.ValidateEnv),
+		ulimits:   runconfigopts.NewUlimitOpt(&ulimits),
+	}
 
-	cmd.Require(flag.Exact, 1)
+	cmd := &cobra.Command{
+		Use:   "build PATH | URL | -",
+		Short: "Build an image from a Dockerfile",
+		Args:  cli.RequiresMinArgs(1),
```
Comment (dnephin) [inline api/client/image/build.go]: `ExactArgs(1)` I think?

Comment (vdemeester) [inline api/client/image/build.go]: Hum That is true :sweat_smile:

**Comment by cpuguy83** [comment]
LGTM

**Comment by LK4D4** [comment]
LGTM

**Pull request merged**